### ISSUE TITLE
Run only first test from Suite

### ISSUE THREAD

**Issue by fraxiu** [testng-team/testng#1639]
### TestNG Version
6.13.1

### Expected behavior
Run all tests

### Actual behavior
Run only first test

### Is the issue reproductible on runner?

- [ ] Shell
- [x] Maven
- [ ] Gradle
- [ ] Ant
- [ ] Eclipse
- [ ] IntelliJ
- [ ] NetBeans

### Test case sample
In TestNG v.6.11 I launched all my tests with maven command: `mvn clean test` without any parameters, and all 242 tests from my suite was launched. 

After upgrade to v.6.13.1, only first test was launched and Suite finished with positive report, bu it contained only one test.

Im using Selenium v.3.8.1, maven-compiler-plugin.version 3.7.0, maven-surefire-plugin.version 2.20.1

My testng.xml contains only package name and all tests from that package were launched in previous version of TestNG. It contains only 9 lines:

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite thread-count="1" name="BrainshareItSuite">
<test name="AllTests" >
    <packages>
        <package name="com.brainshare.faktury.qa.*"/>
    </packages>
  </test>
</suite>
```

**Comment by krmahadevan** [comment]
@fraxiu - I am not able to reproduce this issue on TestNG `6.13.1`. If this is still a problem for you, I request you to please help create a sample project that we can execute to recreate the issue.

Here's what I have.

**Suite xml**

```xml
<?xml version="1.0" encoding="UTF-8"?>
<!DOCTYPE suite SYSTEM "http://testng.org/testng-1.0.dtd">
<suite name="many_methods_suite" verbose="2">
    <test name="many_methods_test">
        <packages>
            <package name="test.github1639.*"/>
        </packages>
    </test>
</suite>
```

The test class was just this but spread across multiple packages

```java
import org.testng.annotations.Test;

public class TestClass {
    @Test
    public void testMethod() {
        System.err.println(getClass().getName() + " says hello !!");
    }
}
```
package structure looks like this

```bash
22:36 $ cd src/test/java/test/github1639/
✔ ~/temp/example/src/test/java/test/github1639
22:36 $ tree .
.
├── five
│   └── TestClass.java
├── four
│   └── TestClass.java
├── one
│   └── TestClass.java
├── three
│   └── TestClass.java
└── two
    └── TestClass.java
```

maven output looks like below

```
[INFO]
[INFO] --- maven-surefire-plugin:2.20.1:test (default-test) @ example ---
[INFO]
[INFO] -------------------------------------------------------
[INFO]  T E S T S
[INFO] -------------------------------------------------------
[INFO] Running TestSuite
...
... TestNG 6.13.1 by Cédric Beust (cedric@beust.com)
...

test.github1639.three.TestClass says hello !!
test.github1639.one.TestClass says hello !!
test.github1639.two.TestClass says hello !!
test.github1639.five.TestClass says hello !!
test.github1639.four.TestClass says hello !!
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0, Time elapsed: 0.394 s - in TestSuite
[INFO]
[INFO] Results:
[INFO]
[INFO] Tests run: 5, Failures: 0, Errors: 0, Skipped: 0
[INFO]
[INFO] ------------------------------------------------------------------------
[INFO] BUILD SUCCESS
[INFO] ------------------------------------------------------------------------
[INFO] Total time: 3.607 s
[INFO] Finished at: 2017-12-21T22:39:52+05:30
[INFO] Final Memory: 23M/309M
[INFO] ------------------------------------------------------------------------
```

**Comment by kamikrazeguy** [comment]
Was there any update on this?

**Comment by krmahadevan** [comment]
@kamikrazeguy - What update are you looking for ?

**Comment by kamikrazeguy** [comment]
we too are running into an issue where only the first test is executed. We are using Maven+Java+TestNG. The TestNG xml is very similar to yours so not sure what is going on.

**Comment by krmahadevan** [comment]
@kamikrazeguy - Can you please share a sample project which we can use to reproduce the problem ?

**Comment by kamikrazeguy** [comment]
I'll see if I can create something for you since the current projects are enterprise based with sensitive data. Thanks for the follow up and let me know what you need as well

**Comment by krmahadevan** [comment]
NO updates from user. Closing this issue. Please comment and add a standalone sample as requested if this is still a problem using the latest released version of TestNG viz., `6.14.3`

**Comment by krmahadevan** [comment]
Please retry using `7.0.0-beta1`. If it's still a problem please share a sample project using which we can reproduce the problem.

**Comment by krmahadevan** [comment]
@fraxiu - Thanks for the kind words. I am just a contributor to TestNG. All credit goes to @cbeust for building TestNG 😄

**Comment by rahulrakesh16** [comment]
Testng Version 6.14.3, 
Issue: TestNg runs only first Test method from each class.  Easily reproducible.

**Comment by krmahadevan** [comment]
Latest released version of TestNG is `7.0.0-beta3`. Please retry using that and post back if it's still a problem

**Comment by fraxiu** [comment]
On wersion 7.0.0-beta3 also everything is fine.

To reproduce I can write, that I have some classes inside folder/directory:

. ├── dirOne │   └── TestClass1.java  │   └── TestClass2.java

and I launch all tests inside that directory using command:

mvn -Dtest=*/dirOne/* test
